src/nbl/asset/utils/CIESProfile.cpp

```diff
@@ -104,8 +104,25 @@ inline std::pair<float, float> CIESProfile::sphericalDirToRadians(const core::ve
 }
 
 template<class ExecutionPolicy>
-core::smart_refctd_ptr<asset::ICPUImageView> CIESProfile::createIESTexture(ExecutionPolicy&& policy, const size_t& width, const size_t& height) const
+core::smart_refctd_ptr<asset::ICPUImageView> CIESProfile::createIESTexture(ExecutionPolicy&& policy, asset::IAssetLoader::IAssetLoaderOverride* aOverride, const std::string key, const float flatten, const size_t width, const size_t height) const
 {
+    const bool inFlattenDomain = flatten >= 0.0 && flatten < 1.0; // [0, 1) range for blend equation, 1 is invalid
+    
+    assert(inFlattenDomain);
+    assert(aOverride);
+
+    const auto aHash = key + "?flatten=" + std::to_string(flatten);
+    asset::IAssetLoader::SAssetLoadContext ctx = { {}, nullptr };
+
+    const asset::IAsset::E_TYPE types[]{ asset::IAsset::ET_IMAGE_VIEW, asset::IAsset::ET_TERMINATING_ZERO };
+    SAssetBundle bundle = aOverride->findCachedAsset(aHash, types, ctx, 0u);
+    {
+        auto contents = bundle.getContents();
+
+        if (!contents.empty() && bundle.getAssetType() == asset::IAsset::ET_IMAGE_VIEW)
+            return core::smart_refctd_ptr_static_cast<asset::ICPUImageView>(*contents.begin());
+    }
```

Comment:
no this goes in the callee of `createIESTexture`

Reply:
resolved in following commit